### PR TITLE
Added config validation

### PR DESCRIPTION
**Pull request by ror6ax**
Fixes situations like https://github.com/jaegertracing/jaeger-client-python/issues/123

### CODE REVIEW

**Comment by ror6ax** [comment]
Ping @yurishkuro

**Comment by yurishkuro** [comment]
Please make sure to sign all commits (see CONTRIBUTING.md)

**Comment by yurishkuro** [comment]
ping @ror6ax

**Comment by ror6ax** [comment]
What should be the entity to enable config checking?

File: tests/test_config.py [hunk 1]
```diff
@@ -100,3 +100,7 @@ def test_propagation(self):
 
         c = Config({'propagation': 'b3'}, service_name='x')
         assert len(c.propagation) == 1
+
+    def test_for_unexpected_config_entries(self):
+        with self.assertRaises(Exception):
+            _ = Config({"unexpected":"value"}, )
```
Comment (yurishkuro) [inline tests/test_config.py]: why is it expected to raise an exception of `'check_config_format'` is not set?

File: jaeger_client/config.py [hunk 2]
```diff
@@ -104,6 +106,25 @@ def __init__(self, config, metrics=None, service_name=None, metrics_factory=None
             logger=logger if self.logging else None,
         )
 
+    def _validate_config(self, config):
+        allowed_keys = ['logging',
```
Comment (yurishkuro) [inline jaeger_client/config.py]: should `'check_config_format'` be here too?

**Comment by yurishkuro** [comment]
> What should be the entity to enable config checking?

So I originally though of the same approach you used, a property in the config, but it has a downside that if that property itself is misspelled, then the config won't be validated. So another approach could be an optional parameter to the Config class constructor:
```
def __init__(self, config, metrics=None, service_name=None, metrics_factory=None, validate=False):
```

**Comment by ror6ax** [comment]
I'm not sure why crossdock test started failing...

**Comment by black-adder** [comment]
From the logs:

jaeger_client/config.py:83:101: E501 line too long (102 > 100 characters)
jaeger_client/config.py:125:30: Q000 Remove bad quotes
jaeger_client/config.py:125:74: Q000 Remove bad quotes
jaeger_client/config.py:125:83: F821 undefined name 'nonexpected_config_keys'
jaeger_client/config.py:125:101: E501 line too long (108 > 100 characters)

**Comment by ror6ax** [comment]
Sorry, that was stupid. Now seems to be working fine.

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**